### PR TITLE
[zh-cn] Init translation of 'deploying our app'

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

Finish translation of 'deploying our app'

### Motivation

Fix up translation in 'Understanding client-side tools' module

### CODE REVIEW

File: files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+- 端到端测试，即访问者点击某样东西，然后发生另一件事。
+- 集成测试，基本上是说“当一个代码块连接到另一个代码块时，它还能工作吗”。
+- 单元测试，即测试小而具体的功能位，看它们是否完成了应该做的事情。
+- [还有更多类型](https://en.wikipedia.org/wiki/Functional_testing)。此外，请参阅我们的[跨浏览器测试模块](/zh-CN/docs/Learn/Tools_and_testing/Cross_browser_testing)，获取大量有用的测试信息。
+
+请记住，测试并不局限于 JavaScript；还可以针对渲染的 DOM、用户交互、CSS 甚至页面外观运行测试。
+
+不过，在这个项目中，我们将创建一个小测试，检查第三方 NASA 数据源，确保其格式正确。如果格式不正确，测试就会失败，导致项目无法上线。做任何其他事情都超出了本模块的范围——测试是一个庞大的主题，确实需要有自己独立的模块。我们希望这部分内容至少能让你意识到测试的必要性，并能播下一粒种子，激发你去学习更多的知识。
+
+虽然本项目的测试不包括测试框架，但正如前端开发领域的所有东西一样，有大量的[框架选项](https://www.npmjs.com/search?q=keywords%3Atesting)。
```
Comment (JasonLamv-t) [inline files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md:232]:
```suggestion
虽然本项目的测试不包括测试框架，但正如前端开发领域的所有东西一样，有大量[可选择的框架](https://www.npmjs.com/search?q=keywords%3Atesting)。
```

File: files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+测试本身并不重要。重要的是如何处理失败或成功。一些部署平台会将特定的测试方法作为其管道的一部分。GitHub 和 GitLab 等产品都支持针对单个提交运行测试。
```
Comment (JasonLamv-t) [inline files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md:234]:
```suggestion
测试本身并不重要。重要的是如何处理失败或成功。一些部署平台会将特定的测试方法作为其流水线的一部分。GitHub 和 GitLab 等产品都支持针对单个提交运行测试。
```

File: files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+由于本项目将部署到 Netlify，而 Netlify 只询问构建命令，因此我们必须将测试作为构建的一部分。如果测试失败，构建也会失败，Netlify 就不会部署。
+
+让我们开始吧。
+
+1. 打开你的 `package.json` 文件。
+2. 找到 `scripts` 成员，更新它以使其包含以下测试和构建命令：
```
Comment (JasonLamv-t) [inline files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md:241]:
```suggestion
2. 找到 `scripts` 属性，更新它以使其包含以下测试和构建命令：
```

File: files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+   git push github main
+   ```
+
+   在某些情况下，你可能想测试构建代码的结果（因为这并不完全是我们编写的原始代码），所以测试可能需要在构建命令之后运行。在开发自己的项目时，你需要考虑所有这些不同的方面。
+
+最后，在推送一分钟左右后，Netlify 将部署项目更新。但前提是它必须通过前面介绍的测试。
+
+## 总结
+
+我们的案例研究示例和本模块就介绍到这里！希望对你有所帮助。虽然要成为客户端工具向导还有很长的路要走，但我们希望本模块能让你迈出了解客户端工具的重要第一步，并有信心学习更多知识和尝试新事物。
```
Comment (JasonLamv-t) [inline files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md:299]:
```suggestion
我们的示例案例研究和本模块就介绍到这里！希望对你有所帮助。虽然要成为客户端工具向导还有很长的路要走，但我们希望本模块能让你迈出了解客户端工具的重要第一步，并有信心学习更多知识和尝试新事物。
```

**Review by JasonLamv-t — CHANGES_REQUESTED** [review]
Some suggestions.

File: files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+  </tbody>
+</table>
+
+## 在部署之前
+
+在项目生命周期的这一部分，可能有大量问题需要解决。因此，必须创建一个工具链，以尽可能少的人工干预来处理这些问题。
+
+下面是这个特殊项目需要考虑的一些事项：
+
+- 生成生产构建：确保文件最小化、分块、应用树状震荡，并确保版本是“cache busted”的。
```
[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md:35]: Should we add "tree shaking" to our glossary for better reference?

[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md:35]: BTW, is there any suggestion for 'cache-busting'?

[the remark this answers is not in the document]
Comment (JasonLamv-t) [reply files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md:35]: > BTW, is there any suggestion for 'cache-busting'?

I think the most direct translation is "缓存破坏", but it doesn't conform to the sentence, so let's make some changes.

```suggestion
- 生成生产构建：确保文件最小化、分块、应用树状震荡，并确保构建版本令[客户端缓存失效](Glossary_of"缓存破坏")。
```

We also need to add the "cache-busting" to the glossary to describe more details.

File: files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+
+虽然 Netlify 提供[拖放部署服务](https://app.netlify.com/drop)，但我们打算每次向 GitHub 仓库推送时都向 Netlify 触发新的部署。
+
+我们鼓励大家在决定自己的构建工具链时，寻找这类互联服务。我们可以提交代码并推送到 GitHub，更新后的代码会自动触发整个构建过程。如果一切顺利，我们就能自动部署实时变更。我们需要执行的*唯一*操作就是最初的“推送”。
+
+不过，我们确实需要设置这些步骤，现在我们就来看看。
+
+## 构建过程
+
+同样，由于我们使用 Parcel 进行开发，因此添加构建选项非常简单。我们可以使用 `npx parcel build src/index.html` 来运行服务器，而不是使用 `npx parcel src/index.html`，这样 Parcel 就会为生产准备好一切，而不仅仅是为了开发和测试而运行。这包括对代码进行最小化和摇树（tree-shaking）处理，以及对文件名进行 cache-busting。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/understanding_client-side_tools/deployment/index.md:57]:
```suggestion
同样，由于我们使用 Parcel 进行开发，因此添加构建选项非常简单。我们可以使用 `npx parcel build src/index.html` 来运行服务器，而不是使用 `npx parcel src/index.html`，这样 Parcel 就会为生产准备好一切，而不仅仅是为了开发和测试而运行。这包括对代码进行最小化和摇树优化处理，以及对文件名进行缓存破坏。
```

**Review by JasonLamv-t — APPROVED** [review]
LGTM. Thank you! @jasonren0403 👍🏼

**Pull request merged**